subt_ign/src/path_tracer.cc

```diff
@@ -0,0 +1,354 @@
+  std::thread playbackThread(std::bind(&Processor::Playback, this, _path));
+
+  this->cv.wait(lock);
+
+  // Create a transport node that uses the default partition.
+  ignition::transport::Node node;
+
+  // Subscribe to the robot pose topic
+  bool subscribed = false;
+  for (int i = 0; i < 5 && !subscribed; ++i)
+  {
+    std::vector<std::string> topics;
+    node.TopicList(topics);
+
+    // Subscribe to the first /dynamic_pose/info topic
+    for (auto const &topic : topics)
+    {
+      if (topic.find("/dynamic_pose/info") != std::string::npos)
+      {
+        // Subscribe to a topic by registering a callback.
+        if (!node.Subscribe(topic, &Processor::Cb, this))
+        {
+          std::cerr << "Error subscribing to topic ["
+            << topic << "]" << std::endl;
+          return;
+        }
+        subscribed = true;
+        break;
+      }
+    }
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+  }
+
+  // Wait for log playback to end.
+  playbackThread.join();
+
+  // Process the events log file.
+  YAML::Node events = YAML::LoadFile(_path + "/events.yml");
+  for (std::size_t i = 0; i < events.size(); ++i)
+  {
+    if (events[i]["type"].as<std::string>() == "artifact_report_attempt")
+    {
+      ignition::math::Vector3d reportedPos;
+
+      // Read the reported pose.
+      std::stringstream stream;
+      stream << events[i]["reported_pose"].as<std::string>();
+      stream >> reportedPos;
+
+      int sec = events[i]["time_sec"].as<int>();
+      std::unique_ptr<ReportData> data(new ReportData);
+      data->type = REPORT;
+      data->pos = reportedPos;
+      data->score = events[i]["points_scored"].as<int>();
+
+      this->logData[sec].push_back(std::move(data));
+    }
+  }
+  // Display all of the artifacts using visual markers.
+  this->DisplayArtifacts();
+
+  // Display all of the poses using visual markers.
+  this->DisplayPoses();
+}
+
+//////////////////////////////////////////////////
+Processor::~Processor()
+{
+  delete this->markerNode;
+  this->markerNode = nullptr;
+}
+
+//////////////////////////////////////////////////
+void Processor::ClearMarkers()
+{
+  ignition::msgs::Marker markerMsg;
+  markerMsg.set_ns("default");
+  markerMsg.set_action(ignition::msgs::Marker::DELETE_ALL);
+  markerNode->Request("/marker", markerMsg);
+  std::this_thread::sleep_for(std::chrono::seconds(1));
+}
+
+//////////////////////////////////////////////////
+void Processor::Playback(std::string _path)
+{
+  std::unique_lock<std::mutex> lock(this->mutex);
+  ignition::transport::log::Playback player(_path + "/state.tlog");
+  bool valid = false;
+  ignition::transport::log::PlaybackHandlePtr handle;
+
+  // Playback all topics
+  const int64_t addTopicResult = player.AddTopic(std::regex(".*"));
+  if (addTopicResult == 0)
+    std::cerr << "No topics to play back\n";
+  else if (addTopicResult < 0)
+    std::cerr << "Failed to advertise topics: " << addTopicResult << "\n";
+  else
```

Comment:
I think we're supposed to have braces around multiple logical statements.

```suggestion
  if (addTopicResult == 0)
  {
    std::cerr << "No topics to play back\n";
  }
  else if (addTopicResult < 0)
  {
    std::cerr << "Failed to advertise topics: " << addTopicResult << "\n";
  }
  else
```

Reply:
Fixed in 4890364ef47509e855e24a38e3fd5e5b3a6da1c5